kuma/wiki/jobs.py

```diff
@@ -44,3 +47,45 @@ def fetch(self, locale):
     def empty(self):
         # the empty result needs to be an empty list instead of None
         return []
+
+
+class DocumentContributorsJob(KumaJob):
+    """
+    Given a wiki document returns a list of contributors that have recently
+    authored revisions.
+
+    We invalidate this when a document is saved only, not when a user account
+    changes given the potential of lots of documents needing to be updated
+    everytime a profile is saved. Instead we accept that some contributor links
+    may be wrong until the cache item's lifetime runs out for this edge case.
+    """
+    lifetime = 60 * 60 * 6
+    refresh_timeout = 30
+
+    def fetch(self, pk):
+        from .models import Document
+        User = get_user_model()
+
+        # first get a list of user ID recently authoring revisions
+        document = Document.objects.get(pk=pk)
+        recent_creator_ids = (document.revisions.order_by('-created')
+                                                .values_list('creator_id',
+                                                             flat=True))
+
+        if not recent_creator_ids:
+            return self.empty()
+
+        # then return the ordered results given the ID list, MySQL only syntax
+        select = collections.OrderedDict([
+            ('ordered_ids',
+             'FIELD(id,%s)' % ','.join(map(str, recent_creator_ids))),
```

Comment:
It's what's recommended in the Django docs since order matters here and we may want to extend it in the future.